1_fetch/src/zip_to_sf.R

```diff
@@ -0,0 +1,12 @@
+# unzip a zipped shapefile into a tempdir, read it as sf, and return the sf object
+zip_to_sf <- function(zipfile) {
+  zippath <- normalizePath(zipfile)
+  unzipdir <- tempfile()
+  dir.create(unzipdir)
+  oldwd <- setwd(unzipdir)
+  on.exit(setwd(oldwd))
```

Comment:
I'm confused why the working directory is set to `tempfile()` upon exiting the function - should that be the current directory instead?

Reply:
`oldwd <- setwd(xx)` sets `oldwd` to the working directory _before_ `setwd(xx)` is applied, so we are indeed setting it back to the current directory on exit. Equivalent code would be
```
oldwd <- getwd()
setwd(unzipdir)
on.exit(setwd(oldwd))
```
I hadn't thought about it much before, but I see how the shorthand syntax is less clear, and I may move to this more explicit syntax in the future.

Reply:
Oh, wow. I've never seen that syntax before.

Reply:
I edited it here.